src/OrchardCore/OrchardCore.Localization.Core/PortableObject/PoParser.cs

```diff
@@ -3,224 +3,232 @@
-        /// Parses a .po file.
-        /// </summary>
-        /// <param name="reader">The <see cref="TextReader"/>.</param>
-        /// <returns>A list of culture records.</returns>
+    /// <summary>
+    /// Parses a .po file.
+    /// </summary>
+    /// <param name="reader">The <see cref="TextReader"/>.</param>
+    /// <returns>A list of culture records.</returns>
 #pragma warning disable CA1822 // Mark members as static
-        public IEnumerable<CultureDictionaryRecord> Parse(TextReader reader)
+    public async IAsyncEnumerable<CultureDictionaryRecord> ParseAsync(TextReader reader)
 #pragma warning restore CA1822 // Mark members as static
+    {
+        var entryBuilder = new DictionaryRecordBuilder();
+        string line;
+        while ((line = await reader.ReadLineAsync()) != null)
         {
-            var entryBuilder = new DictionaryRecordBuilder();
-            string line;
-            while ((line = reader.ReadLine()) != null)
-            {
-                (var context, var content) = ParseLine(line);
+            (var context, var content) = ParseLine(line);
 
-                if (context == PoContext.Other)
-                {
-                    continue;
-                }
-
-                // msgid or msgctxt are first lines of the entry. If builder contains valid entry return it and start building a new one.
-                if ((context == PoContext.MessageId || context == PoContext.MessageContext) && entryBuilder.ShouldFlushRecord)
-                {
-                    yield return entryBuilder.BuildRecordAndReset();
-                }
-
-                entryBuilder.Set(context, content);
+            if (context == PoContext.Other)
+            {
+                continue;
             }
 
-            if (entryBuilder.ShouldFlushRecord)
+            // msgid or msgctxt are first lines of the entry. If builder contains valid entry return it and start building a new one.
+            if ((context == PoContext.MessageId || context == PoContext.MessageContext) && entryBuilder.ShouldFlushRecord)
             {
                 yield return entryBuilder.BuildRecordAndReset();
             }
+
+            entryBuilder.Set(context, content);
         }
 
-        private static string Unescape(string str)
+        if (entryBuilder.ShouldFlushRecord)
+        {
+            yield return entryBuilder.BuildRecordAndReset();
+        }
+    }
+
+    private static string Unescape(string str)
+    {
+        StringBuilder sb = null;
+        var escaped = false;
+        for (var i = 0; i < str.Length; i++)
         {
-            StringBuilder sb = null;
-            var escaped = false;
-            for (var i = 0; i < str.Length; i++)
+            var c = str[i];
+            if (escaped)
             {
-                var c = str[i];
-                if (escaped)
+                if (sb == null)
                 {
-                    if (sb == null)
+                    sb = new StringBuilder(str.Length);
```

Comment:
ZStringBuilder?

Reply:
I will create another PR for that because it might need extra testing and benchmarking